src/Take.Elephant.Sql/ExpirableKeySqlMap.ExpirationDatabaseDriverDecorator.cs

```diff
@@ -63,7 +66,19 @@ public string GetSqlStatementTemplate(SqlStatement sqlStatement)
 
             private static string InjectSqlFilter(string sql, string filter)
             {
-                if (sql.Contains("ORDER BY")) return sql.Replace("ORDER BY", $"{filter} ORDER BY");
+                if (sql.Contains("ORDER BY"))
```

Comment:
It seems that we can use a single rule that will work for any case: check for "WHERE {filter}" or even just "{filter}", and then replace by $"WHERE {{filter}} {expirationFilter} " (I have renamed the parameter to disambiguate with filter inside Sql templates.

Reply:
Oh yea, nice catch!